R/Design-methods.R

```diff
@@ -131,6 +131,21 @@ getResultList <- function(fun,
         varlist = ls(.GlobalEnv)
       )
 
+      # load user extensions
+      tryCatch(
+        {
+          if (exists("crmpack_extensions") == TRUE) {
```

Comment:
instead of doing this with a function definition as a global variable - I think it would be better to do it via a global option (https://stat.ethz.ch/R-manual/R-devel/library/base/html/options.html)

Reply:
I am not sure if I understand the comment. Do you mean to make sure that it is really a function and not just an object? I am not sure how to this via the option statement.

Reply:
Sorry that I did not give an example.
Try out this to see the mechanism:
```r
options(extensions = function(x) x + 2)
getOption("extensions")
```

Reply:
Thanks @0liver0815 , ok seems not super easy, but here would be my proposal:

```r
library(parallel)
# Globally, work with an option. This is safer than a global function name.
options(crmpack_extensions = function() "bla")
# Now inside getResultList(), we can check the global option and load into
# a local function.
fun <- getOption("crmpack_extensions")
cl <- parallel::makeCluster(2)
# We can export the local function to the cluster.
parallel::clusterExport(cl, "fun", environment(fun))
# And then evaluate it.
parallel::clusterEvalQ(cl, fun())
```

would that work? main point here is to avoid the need for a globally defined function of a given name, and replace it by a global option holding the function instead.

Reply:
I think I was to quick with my assessment. After more in-depth testing, the code above do not run within the function `getResultList`.  The reasons remains unclear to me. The problem is that when `clusterEvalQ` is invoked, the function is unknown to `clusterEvalQ`, even though it is available. However, it seems that `clusterCall` solves the problem.